JS/Lesson34/onur/my-app/src/components/NewRecipe/index.jsx

```diff
@@ -22,56 +23,195 @@ export const NewRecipe = () => {
+				<textarea
+					className='form-input'
+					id='strInstructions'
+					{...register('strInstructions', { required: true, minLength: 10 })}
+				></textarea>
+				{errors.strInstructions && (
+					<span className='input-error'>Instructions are required</span>
+				)}
+
+				<label className='form-label' htmlFor="strMealThumb">Meal Thumb :</label>
+				<input type="text" id='strMealThumb' className='form-input' {...register("strMealThumb", {
+					required: true,
+					pattern: /\.(jpg|jpeg)$/
+				})} />
+				{errors.strMealThumb && (
+					<span className='input-error'>Please enter a valid URL</span>
+				)}
+
+				<label className='form-label' htmlFor='strTags'>Tag Name :</label>
+				<input
+					className='form-input'
+					id='strTags'
+					type='text'
+					{...register('strTags', { required: true, minLength: 2 })}
+				/>
+				{errors.strTags && (
+					<span className='input-error'>Tag name is required</span>
+				)}
+
+				<label className='form-label' htmlFor="strYoutube">Youtube video :</label>
+				<input type="text" id='strYoutube' className='form-input' {...register("strYoutube", {
+					required: true,
+					pattern: /^(https?:\/\/)?(www\.)?(youtube\.com\/(?:[^\/\n\s]+\/\S+\/|(?:v|e(?:mbed)?)\/|\S*?[?&]v=)|youtu\.be\/)([a-zA-Z0-9_-]{11})/
+				})} />
+				{errors.strYoutube && (
+					<span className='input-error'>Please enter a valid Youtube URL</span>
+				)}
+
+				<label className='form-label'>Ingredients : </label>
+				<textarea className='form-input' placeholder='Please separate by commos' {...register("ingredients", { required: true, minLength: 1 })} />
+				{errors.ingredients && (
+					<span className='input-error'>Ingredients are required</span>
+				)}
+
+				<label className='form-label'>Measures : </label>
+				<textarea className='form-input' placeholder='Please separate by commos' {...register("measures", { required: true, minLength: 1 })} />
+				{errors.measures && (
+					<span className='input-error'>Measures is required</span>
+				)}
+
+				<label className='form-label' htmlFor="strSource">Source :</label>
+				<input className='form-input' type="text" id='strSource' {...register("strSource", {
+					required: true,
+					pattern: /^https:\/\//
+				})} />
+				{errors.strSource && (
+					<span className='input-error'>Please enter a valid  address</span>
+				)}
+
+				<label className='form-label' htmlFor="strImageSource">Image source :</label>
+				<input className='form-input' type="text" id='strImageSource' {...register("strImageSource", { pattern: /^https:\/\// })} />
+
+				<label className='form-label' htmlFor="strCreativeCommonsConfirmed">Creative Commons :</label>
+				<input className='form-input' type="text" id='strCreativeCommonsConfirmed' {...register("strCreativeCommonsConfirmed")} />
+
+				<label className='form-label' htmlFor="dateModified">Date Modified :</label>
+				<input className='form-input' type="text" id='dateModified' {...register("dateModified")} />
```

Comment:
Values like date modified, date created, etc. are usually created programatically. So when the form is submitted, create a new data and assign it to your own variable for the form data before sending the data to server or using it in some other way.

Reply:
And remove the input element for the date :)